app/api/v2/admin/profiles.rb

```diff
@@ -36,14 +36,16 @@ class Profiles < Grape::API
             optional :address, type: String
             optional :postcode, type: String
             optional :city, type: String
+            optional :state, type: String
             optional :country, type: String
             optional :metadata, type: String, desc: 'Any additional key: value pairs in json string format'
           end
 
           put do
-            target_profile = User.find_by(uid: params[:uid])&.profile
-            return error!({ errors: ['admin.profiles.doesnt_exist'] }, 404) if target_profile.nil?
+            target_profile = User.find_by(uid: params[:uid])&.profiles&.find_by(state: 'drafted')
```

Comment:
better make model methods we can understand like 
latest_draft
latest_approved
something like that
avoid 1 liners it's evil